gemini/sample-apps/fashion-trends-prediction/app/data_prep/driver.py

```diff
@@ -0,0 +1,332 @@
+        influencers = getInfluencers(country_name)
+
+        for influencer in influencers[: scrape_size["num_influencers"]]:
+            print(influencer)
+
+            if (
+                influencer not in scraped_influencers
+            ):  # this influencer was not in the list of any other country till now
+                scraped_influencers.add(influencer)
+                # the influencer was there in the previous run
+                if influencer in saved["global"]:
+                    posts = getPosts(
+                        influencer,
+                        saved["global"][influencer],
+                        scrape_size["num_posts"],
+                        cookies,
+                        model="Gemini",
+                    )
+                else:
+                    posts = getPosts(
+                        influencer,
+                        [],
+                        scrape_size["num_posts"],
+                        cookies,
+                        model="Gemini",
+                    )
+
+                temp_map[influencer] = posts
+
+            time.sleep(1)
+
+            # save data after running for each influencer
+            with open(data_path, "w") as outfile:
+                json.dump(temp_map, outfile)
+
+    # delete the influencers that are not in the list of any country anymore
+    saved["global"] = {k: v for k, v in temp_map.items() if k in scraped_influencers}
+
+    with open(data_path, "w") as outfile:
+        json.dump(saved, outfile)
+
+
+def periodic_extraction(scrape_size, first_period):
+    """Extracts data from Instagram periodically and saves it in a JSON file.
+
+    Args:
+            scrape_size (dict): A dictionary containing the number of countries, influencers, and posts to scrape.
+            first_period (bool): A boolean indicating whether this is the first period.
+
+    """
+
+    if first_period:
+        saved = {}
+        saved["global"] = {}
+    else:
+        with open(data_path, "r") as f:
+            saved = json.load(f)
+
+    # first scrape is set to True when you are running for the first time in an extraction period
+    extract_json_influencer_wise(
+        scrape_size=scrape_size, first_scrape=True, saved=saved
+    )
+
+
+def create_final_data(scrape_size):
+    """Creates a final data file by combining data from Instagram and Vogue.
+
+    Args:
+            scrape_size (dict): A dictionary containing the number of countries, influencers, and posts to scrape.
+
+    """
+
+    with open(data_path, "r") as f:
+        saved = json.load(f)
+
+    saved["finaldata"] = {}
+    for country_name in country_names[: scrape_size["num_countries"]]:
+        print(country_name)
+        saved["finaldata"][country_name] = {}
+
+        # the top influencers of that country right now
+        influencers = getInfluencers(country_name)
+
+        for influencer in influencers[: scrape_size["num_influencers"]]:
```

Comment:
This level nesting makes the code difficult to read. Can you refactor this?

Reply:
Refactored